include/irr/asset/IMeshPacker.h

```diff
@@ -0,0 +1,123 @@
+#ifndef __IRR_I_MESH_PACKER_H_INCLUDED__
+#define __IRR_I_MESH_PACKER_H_INCLUDED__
+
+namespace irr
+{
+namespace asset
+{
+
+namespace meshPackerUtil
+{
+    struct AllocationParams
+    {
+        size_t indexBuffSupportedSizeInBytes                 = 2147483648ull;      /*  2GB*/
+        size_t vertexBuffSupportedSizeInBytes                = 2147483648ull;      /*  2GB*/
+        size_t perInstanceVertexBuffSupportedSizeInBytes     = 33554432ull;        /* 32MB*/
+        size_t MDIDataBuffSupportedSizeInBytes               = 2147483648ull;      /*  2GB*/
+        size_t vertexBufferMinAllocSize                      = 512ull;             /* 512B*/
+        size_t indexBufferMinAllocSize                       = 256ull;             /* 256B*/
+        size_t perInstanceVertexBufferMinAllocSize           = 32ull;              /*  32B*/
+        size_t MDIDataBuffMinAllocSize                       = 32ull;              /*  32B*/
```

Comment:
your address spaces are in bytes, but your minimum allocation sizes are in Units of `sizeof(typeof(data))`

For the sake of consistency, maybe switch from `SizeInBytes` to `Count` and divide all your defaults by `sizeof(uint32_t)` (everything else) and `sizeof(uint16_t)` (indices) where appropriate

Reply:
STILL

Reply:
only exception is vertex data, since that doesn't really have a fixed sizeof

Reply:
It will be changed that in next commit